### DISCUSSION TITLE
Question regarding `NeighborLoader` sampling

### DISCUSSION THREAD

**Discussion by skalyanb** [pyg-team/pytorch_geometric#3630] [General]
I have a question regarding the usage of the `NeighborLoader`.  I have a non-tensor node property in my graph. So my `Data` object looks something like this:
```
Data(x=[10000, 200], edge_index=[2, 822330], y=[10000, 1], prop=[10000], train_mask=[1000], n_id=[10000])
```
where the `prop` is a list of custom objects. When I use `NeighborLoader` to get a batch, the `prop` is untouched.
```
> loader = NeighborLoader (g, num_neighbors=[-1], batch_size=32) 
> batch = next(iter(loader))
> batch
  Data(x=[600, 200], edge_index=[2, 5000], y=[529, 1], prop=[10000], train_mask=[600], n_id=[600], batch_size=32)
```

I can use the `n_id` to retrieve the `prop` for the nodes involved in the batch, but is there a way to tell the `NeighborLoader` to do this automatically?

**Comment by rusty1s** [comment]
Yeah, this is currently not supported (any non-tensor will be simply passed through). I'm sorry for that :(

We will have to implement logic for that here: https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/utils.py#L59-L74, *e.g.*, by filtering via `index` in case we encounter a list of length `num_nodes`. Let me know if you have interest in contributing this feature :)

**Reply by skalyanb** [reply]
I can contribute to this feature. Can I slack you in the PyG channel regarding this?

**Reply by rusty1s** [reply]
Sure, sounds awesome :)
